web/src/inbox_ui.js

```diff
@@ -864,7 +864,53 @@ function center_focus_if_offscreen() {
     $elt[0].scrollIntoView({block: "center"});
 }
 
+function move_focus_to_visible_area() {
+    // Focus on the row below inbox filters if the focused
+    // row is not visible.
+    if (!is_list_focused()) {
+        return;
+    }
+
+    const $all_rows = get_all_rows();
+    if ($all_rows.length <= 3) {
+        // No need to process anything if there are only a few rows.
+        return;
+    }
+
+    if (row_focus >= $all_rows.length) {
+        row_focus = $all_rows.length - 1;
+        revive_current_focus();
+    }
+
+    const elt_pos = $all_rows[row_focus].getBoundingClientRect();
+    if (is_element_visible(elt_pos)) {
+        return;
+    }
+
+    const INBOX_ROW_HEIGHT = 30;
```

Comment:
Can we make this calculation not hardcoded somehow? Not sure if that height exists in a CSS variable or something.

Reply:
I guess it probably doesn't matter very much if this is precisely accurate (we don't multiply it by a number of rows or anything); so I'm going to merge this and get it on chat.zulip.org and we can see if there's things that feel off in actual use.